File: kp3.js
var http = require('http')
var urlp = require("url");
var qs = require('querystring');

var data = {}

var server = http.createServer(function (request, response) {
	response.writeHead(200, {'Content-Type': 'text/plain'});
	var url = urlp.parse(request.url);
	var query = qs.parse(url.query);
	if (url.pathname == '/get') {
		response.end('value=' + data[query['key']]);
	} else if (url.pathname == '/set') {
		data[query['key']] = query['value'];
		response.end('ok');
	}
});

server.listen(8080);

File: kp3.py
from wsgiref.simple_server import make_server
from urllib2 import urlparse

data = {}

def serve(env, start_response):
	path = env['PATH_INFO']
	args = urlparse.parse_qs(env['QUERY_STRING'])
	key = args['key'][0]

	start_response('200 OK', [('Content-type', 'text/plain')])

	if path == '/get':
		return data[key]
	if path == '/set':
		value = args['value'][0]
		data[key] = value
		return ['ok']
	return ['unknown error']

make_server('', 8080, serve).serve_forever()
